src/search/pdbs/pattern_collection_generator_single_cegar.cc

```diff
@@ -0,0 +1,70 @@
+#include "pattern_collection_generator_single_cegar.h"
+
+#include "cegar.h"
+#include "utils.h"
+
+#include "../option_parser.h"
+#include "../plugin.h"
+
+#include "../utils/markup.h"
+#include "../utils/rng_options.h"
+
+using namespace std;
+
+namespace pdbs {
+PatternCollectionGeneratorSingleCegar::PatternCollectionGeneratorSingleCegar(
+    const options::Options &opts)
+    : max_pdb_size(opts.get<int>("max_pdb_size")),
+      max_collection_size(opts.get<int>("max_collection_size")),
+      wildcard_plans(opts.get<bool>("wildcard_plans")),
+      max_time(opts.get<double>("max_time")),
+      verbosity(opts.get<utils::Verbosity>("verbosity")),
+      rng(utils::parse_rng_from_options(opts)) {
+}
+
+PatternCollectionInformation PatternCollectionGeneratorSingleCegar::generate(
+    const shared_ptr<AbstractTask> &task) {
+    // Store the set of goals in random order.
+    TaskProxy task_proxy(*task);
+    vector<FactPair> goals = get_goals_in_random_order(task_proxy, rng);
+
+    CEGAR cegar(
+        max_pdb_size,
+        max_collection_size,
+        wildcard_plans,
+        max_time,
+        verbosity,
+        rng,
+        task,
+        move(goals));
+    return cegar.compute_pattern_collection();
+}
+
+static shared_ptr<PatternCollectionGenerator> _parse(
+    options::OptionParser &parser) {
+    parser.document_synopsis(
+        "Single CEGAR",
+        "This pattern collection generator implements the single CEGAR algorithm "
+        "described in the paper" + utils::format_conference_reference(
+            {"Alexander Rovner", "Silvan Sievers", "Malte Helmert"},
+            "Counterexample-Guided Abstraction Refinement for Pattern Selection "
+            "in Optimal Classical Planning",
+            "https://ai.dmi.unibas.ch/papers/rovner-et-al-icaps2019.pdf",
+            "Proceedings of the 29th International Conference on Automated "
+            "Planning and Scheduling (ICAPS 2019)",
+            "362-367",
+            "AAAI Press",
+            "2019"));
+    add_cegar_options_to_parser(parser);
+    utils::add_verbosity_option_to_parser(parser);
+    utils::add_rng_options(parser);
+
+    Options opts = parser.parse();
+    if (parser.dry_run())
+        return nullptr;
+
+    return make_shared<PatternCollectionGeneratorSingleCegar>(opts);
+}
+
+static Plugin<PatternCollectionGenerator> _plugin("single_cegar", _parse);
```

Comment:
I would prefer to argue that the existing `cegar()` plugin could also be renamed to be more specific :) It does compute an additive Cartesian heuristic, doesn't it?.

Furthermore, the whole directory `cegar` is no longer the unique CEGAR in Fast Downward. I actually wondered if it wouldn't make more sense to name the directory `cartesian_abstractions` or alike because I already thought that the presence of `cegar.h/cc` in `pdbs` can now be confusing since there is also a directory `cegar`.

Reply:
I'd only change existing user-facing options if it's really necessary. I agree that the "cegar" directory name is not ideal, but I think changing the name has more cons than pros. Renaming it will make merges more difficult and I think the two occurences of "cegar" are separated enough by the fact that one resides in the "pdbs" directory.